Feature/add selectors documentation (#10035)
meshery/meshery: docs/pages/concepts/logical/relationships.md

You can reference and search the full set of registered relationships in Meshery UI --> Setttings --> Registry

> ## Selectors in Meshery

Reviewer: ```suggestion
## Selectors in Relationships
```

> Selectors in Meshery are a powerful tool for defining relationships between different components within the service mesh environment. They are organized as an array, providing a flexible and reusable way to specify conditions for relationships.

Reviewer: ```suggestion
Selectors in Meshery are a powerful tool for defining relationships between different components. Selectors are organized in an array, providing a flexible way to specify match conditions and  corresponding actions. Multiple sets of selectors may be defined in order to precisely match and not match specific components and their attributes. ```

### Structure of Selectors

Selectors consist of an array, and each item in the array contains the following fields:

- `[from, to]`: Defines the relationship between components.
- `[to, from]`: Specifies additional conditions for the relationship.

These pairs establish hierarchical inventory relationships, and the policy and visual paradigm remain consistent across different components. To avoid ambiguity when defining relationships, selectors act as internal scopes within a relationship.

### Example Selectors

Selectors can be applied to various components, enabling a wide range of relationship definitions. Here are some examples:

1. **WASMFilter - EnvoyFilter**
2. **ConfigMap - Pod**
3. **ConfigMap - Deployment**

> ### Hierarchical Relationships

Reviewer: ```suggestion
## Types of Relationships

### Hierarchical Relationships
```

> When defining relationships that involve a large number of combinations between from and to, selectors provide a mechanism to organize and manage these relationships hierarchically. This prevents the need for crafting complex deny attributes and facilitates easier maintenance.

Reviewer: ```suggestion
When defining relationships that involve a large number of combinations between `from` and `to`, selectors provide a mechanism to organize and manage these relationships hierarchically. This prevents the need for crafting complex deny attributes and facilitates easier maintenance.
```

@MUzairS15, please expand upon this phrasing: "....manage these relationships hierarchically." Is this accurate? "Hierarchically" isn't the word that best describes how selectors and sets of selectors augment their evaluation. Please update.

You can reference and search the full set of registered relationships in Meshery UI --> Setttings --> Registry

## Selectors in Relationships

Selectors in Meshery are a powerful tool for defining relationships between different components. Selectors are organized in an array, providing a flexible way to specify match conditions and  corresponding actions. Multiple sets of selectors may be defined in order to precisely match and not match specific components and their attributes. ```

### Structure of Selectors

Selectors consist of an array, and each item in the array contains the following fields:

- `[from, to]`: Defines the relationship between components.
- `[to, from]`: Specifies additional conditions for the relationship.

> These pairs establish hierarchical inventory relationships, and the policy and visual paradigm remain consistent across different components. To avoid ambiguity when defining relationships, selectors act as internal scopes within a relationship.

Author: It was misplaced. It is supposed to be after the `Example` section. Corrected. ✅

### Example Selectors

Selectors can be applied to various components, enabling a wide range of relationship definitions. Here are some examples:

1. **WASMFilter - EnvoyFilter**
2. **ConfigMap - Pod**

> 3. **ConfigMap - Deployment**

Author: Done. ✅
